android/src/main/java/com/swmansion/rnscreens/ScreenTransitionProgressEvent.java

```diff
@@ -0,0 +1,41 @@
+package com.swmansion.rnscreens;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.Event;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+
+public class ScreenTransitionProgressEvent extends Event<ScreenAppearEvent> {
+
+  public static final String EVENT_NAME = "topTransitionProgress";
+
+  private final float mProgress;
+  private final boolean mClosing;
+  private final short mCoalescingKey;
+
+  public ScreenTransitionProgressEvent(int viewId, float progress, boolean closing, short coalescingKey) {
+    super(viewId);
+    mProgress = progress;
+    mClosing = closing;
+    mCoalescingKey = coalescingKey;
+  }
+
+  @Override
+  public String getEventName() {
+    return EVENT_NAME;
+  }
+
+  @Override
+  public short getCoalescingKey() {
+    // All events for a given view can be coalesced.
+    return mCoalescingKey;
```

Comment:
if you want all events for single view to be coalesced you should just return the same key here. I think that what would be bettere here actually would be to make the key depend on `mClosing` such that we don't coalesce closing and opening events

Reply:
This comment should not be there, it slipped in when I copied the code regarding this. We already increment the key on closing and opening events, as mentioned in the https://github.com/software-mansion/react-native-screens/pull/890#discussion_r646477369.